.github/workflows/drivers.yml

```
@@ -0,0 +1,233 @@
+              continue
+            fi
+
+            tasks="$(wc -l < "/tmp/tasks/${builder}/${{ matrix.builder }}/all")"
+
+            if ((tasks)); then
+              builders+=("${builder}")
+            fi
+          done
+
+          echo "builders=${builders[*]}" >> "$GITHUB_OUTPUT"
+
+      - name: Build builders
+        if: ${{ steps.required-builders.outputs.builders != '' }}
+        run: |
+          for builder in ${{ steps.required-builders.outputs.builders }}; do
+            docker build --tag "${builder}:latest" \
+              -f "${BUILDERS_DIR}/${builder}.Dockerfile" \
+              ${{ github.workspace }}/kernel-modules/build/
+          done
+
+      - name: Build drivers
+        if: ${{ steps.required-builders.outputs.builders != '' }}
+        run: |
+          mkdir -p /tmp/{output,bundles,FAILURES}
+
+          # this is required for GHA to upload a build failures artifact when
+          # no build fails
+          touch /tmp/FAILURES/.dummy
+
+          for builder in ${{ steps.required-builders.outputs.builders }}; do
+            # Download bundles for current builder
+            awk '{ print "gs://collector-kernel-bundles-public/bundle-"$1".tgz" }' "/tmp/tasks/${builder}/${{ matrix.builder }}/all" |
+              sort | uniq | gsutil -m cp -I /tmp/bundles
+
+            docker run --rm -i \
+              -v /tmp/tasks:/tasks:ro \
+              -v /tmp/kobuild-tmp/versions-src:/kobuild-tmp/versions-src \
+              -v /tmp/output:/kernel-modules \
+              -v /tmp/bundles:/bundles:ro \
+              -v /tmp/FAILURES:/FAILURES \
+              -e DOCKERIZED=1 \
+              --name "${builder}" \
+              "${builder}:latest" < "/tmp/tasks/${builder}/${{ matrix.builder }}/all"
+
+            rm -rf /tmp/bundles/*
+          done
+
+      - name: Store built drivers
+        uses: actions/upload-artifact@v3
+        with:
+          name: built-drivers
+          path: /tmp/output
+          if-no-files-found: ignore
+          retention-days: 1
+
+      - name: Store build failures
+        uses: actions/upload-artifact@v3
+        with:
+          name: driver-build-failures
+          path: /tmp/FAILURES
+          if-no-files-found: ignore
+          retention-days: 1
+
+  upload-drivers:
+    runs-on: ubuntu-latest
+    needs: build-drivers
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Restore built drivers
+        uses: actions/download-artifact@v3
+        with:
+          name: built-drivers
+          path: /tmp/output/
+
+      - name: Choose destination bucket
+        id: gcp-bucket
+        run: |
+          if [[ "${{ github.event_name == 'pull_request' }}" == "true" ]]; then
+            echo "gcp-bucket=mauro-drivers-test/pr-builds/${HEAD_REF}/${{ github.run_id }}" >> "$GITHUB_OUTPUT"
+          else
+            echo "gcp-bucket=mauro-drivers-test/master" >> "$GITHUB_OUTPUT"
```

You're using the workflow from `main`, which will be triggered for `master` & `release-*` branches. Consider this for destination bucket on branch builds: 
```suggestion
            echo "gcp-bucket=mauro-drivers-test/${{ github.ref_name }}" >> "$GITHUB_OUTPUT"
```

https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

---

This is actually just for testing purposes, once we move to using only GHA the bucket for the master branch is completely different from the one used for staging. I just used a `master` subdirectory because I'm too lazy to go setup 2 separate buckets 😅

---

Just realized, we are fine having tag branches push drivers to the same bucket master pushes too, but we'll probably make it so tags don't build drivers at all, since they should be using the ones built on master anyways.